common/util.go

```diff
@@ -700,3 +703,14 @@ func ValidateUTF8String(fieldName string, strValue string) error {
 	}
 	return nil
 }
+
+// DiscardUnknownProto discards unknown fields in a proto message.
+func DiscardUnknownProto(m proto.Message) error {
+	return protorange.Range(m.ProtoReflect(), func(values protopath.Values) error {
```

Comment:
This is copied from https://pkg.go.dev/google.golang.org/protobuf@v1.34.2/reflect/protorange#example-package-DiscardUnknown

There is a [DiscardUnknown](https://pkg.go.dev/github.com/golang/protobuf/proto#DiscardUnknown) method in the deprecated `github.com/golang/protobuf` library. But it's no longer available in `google.golang.org/protobuf`